### PR TITLE
tests: add new (failing) fde-on-classic nested test

### PR DESCRIPTION
**Pull request by mvo5**
This is a (failing) nested test that will create a VM that
contains a classic system that contains ubuntu-{boot,save,data}
partitions. On ubuntu-data it contains a classic root filesystem
that contains debs/apt.

This is a draft for now as it will not work but it should be a good
basis for further work. It's also known to not work hence there
is a "skip-spread" for now.

### CODE REVIEW

File: tests/nested/manual/fde-on-classic/mk-image.sh [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    IMG="$(readlink -f "$1")"
+    CACHE="$(readlink -f "$2")"
+    MNT="$(readlink -f "$3")"
+    local KERNEL_SNAP
+    KERNEL_SNAP=$(find "$CACHE" -maxdepth 1 -name 'pc-kernel_*.snap' -printf "%f\n")
+
+    mkdir -p "$MNT"
+    sudo kpartx -av "$IMG"
+    
+    loop=$(sudo kpartx -l "$IMG" |tr -d " " | cut -f1 -d:|sed 's/..$//'|head -1)
```
Comment (valentindavid) [inline tests/nested/manual/fde-on-classic/mk-image.sh:178]: loop="$(sudo losetup -P --show -f "${IMG}")"

File: tests/nested/manual/fde-on-classic/mk-image.sh [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+    loop_esp="${loop}"p2
+    loop_boot="${loop}"p3
+    loop_save="${loop}"p4
+    loop_data="${loop}"p5
+    # XXX: on a real UC device this the ESP is "ubuntu-seed"
+    sudo mkfs.vfat /dev/mapper/"$loop_esp"
```
Comment (valentindavid) [inline tests/nested/manual/fde-on-classic/mk-image.sh:184]: Note that `mkfs.vfat` is a backward compatibility symlink and might disappear one day. The correct name is `mkfs.fat`. See https://github.com/dosfstools/dosfstools/blob/master/src/Makefile.am

File: tests/nested/manual/fde-on-classic/mk-image.sh [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+    echo "grub will chainload from ESP to ubuntu-boot"
+    echo "there press ESC and add 'dangerous rd.systemd.debug-shell=1' after kernel.efi"
+}
+
+main() {
+    BOOT_IMG="${1:-./boot.img}"
+    CACHE_DIR="${2:-./cache}"
+    MNT_DIR="${2:-./mnt}"
+    # shellcheck disable=SC2064
+    trap "cleanup \"$CACHE_DIR\" \"$MNT_DIR\"" EXIT INT
```
Comment (valentindavid) [inline tests/nested/manual/fde-on-classic/mk-image.sh:317]: First parameter should be `${BOOT_IMG}` I think.

File: tests/nested/manual/fde-on-classic/mk-image.sh [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+    any:
+      match:
+        name: e*
+      dhcp4: true
+  version: 2
+EOF
+    sudo cp "$CACHE"/00-ethernet.yaml "$DESTDIR"/etc/netplan
+
+    # ensure we can login
+    sudo chroot "$DESTDIR" adduser --disabled-password --gecos "" ubuntu
```
Comment (valentindavid) [inline tests/nested/manual/fde-on-classic/mk-image.sh:103]: Not to depend on host `PATH`, use `/usr/sbin/adduser`.

File: tests/nested/manual/fde-on-classic/mk-image.sh [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+    echo -e "ubuntu\nubuntu" | sudo chroot "$DESTDIR" passwd ubuntu
```
Comment (valentindavid) [inline tests/nested/manual/fde-on-classic/mk-image.sh:104]: `/usr/bin/passwd`

File: tests/nested/manual/fde-on-classic/mk-image.sh [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+    echo "ubuntu ALL=(ALL) NOPASSWD:ALL" | sudo tee -a "$DESTDIR"/etc/sudoers
+
+    # XXX set password for root user
+    sudo chroot "$DESTDIR" sh -c 'echo root:root | chpasswd'
```
Comment (valentindavid) [inline tests/nested/manual/fde-on-classic/mk-image.sh:108]: `/bin/sh`

File: tests/nested/manual/fde-on-classic/mk-image.sh (hunk 5 above)
Comment (valentindavid) [reply tests/nested/manual/fde-on-classic/mk-image.sh:104]: and `/usr/sbin/chpasswd`

File: tests/nested/manual/fde-on-classic/mk-image.sh [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+    sudo sh -c "echo \"PermitRootLogin yes\nPasswordAuthentication yes\" >> $DESTDIR/etc/ssh/sshd_config"
```
[the remark this answers is not in the document]
Comment (alfonsosanchezbeato) [reply tests/nested/manual/fde-on-classic/mk-image.sh]: Maybe it was not getting dash echo built-in in your case. Anyway, changed to using sudo tee, it is more readable indeed.

**Comment by alfonsosanchezbeato** [comment]
FTR I have cheery-picked these commits in https://github.com/snapcore/snapd/pull/11974

**Comment by mvo5** [comment]
I'm closing this because it's now part of #11974

**Pull request closed without merging**